controllers/api.controllers.js

```diff
@@ -38,12 +39,33 @@ exports.getArticles = (req, res, next) => {
 };
 
 exports.getCommentFromArticleID = (req, res, next) => {
-    const { article_id } = req.params;
-    selectCommentFromArticleID(article_id)
+  const { article_id } = req.params;
+  selectCommentFromArticleID(article_id)
     .then((comments) => {
-        res.status(200).send({ comments });
+      res.status(200).send({ comments });
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
+
+exports.postNewComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+
+  if (username === undefined) {
+    return next({ status: 400, msg: "username is required" });
```

Comment:
not needed, non-existent username will result in psql error which should be caught in error handling middleware in your app.